### DISCUSSION TITLE
[FF7] When using OpenGL, no 3d models are shown

### DISCUSSION THREAD

**Discussion by malvarenga123** [julianxhokaxhiu/FFNx#681] [General]
The game loads, but none of the 3d characters are shown. Anything in 2D works ok.
The log is filled with the following error:
`[00000318] ERROR: [Device Lost] glDrawElementsInstanced(prim.m_type , numIndices , indexFormat , (void*)(uintptr_t)(draw.m_startIndex*indexSize) , draw.m_numInstances ); GL error 0x502: GL_INVALID_OPERATION`
which you can see here: https://pastebin.com/EcXGtPTm

**Comment by julianxhokaxhiu** [comment]
Are you using FFNx vanilla? Looking at your Specs, your GPU looks very much capable of using DirectX 11 so not sure why you're going for OpenGL. Are you using Windows 7 for a reason?

**Comment by malvarenga123** [comment]
@julianxhokaxhiu yes, this happens with FFNx vanilla.
I'm actually using Vulkan, but I wanted to report this error with OpenGL, because there shouldn't be any reason OpenGL could not work as this GPU should support v4.6.
DX11 is basically unplayable because innumerous artifacts show up and many parts of the screen keep blinking or go black. Using the menu is almost impossible with DX11.
AMD has moved the GPU to legacy mode, so the latest driver is the [same for Windows 7/8/10/11](https://www.amd.com/en/support/graphics/amd-radeon-r9-series/amd-radeon-r9-200-series/amd-radeon-r9-280x).

**Reply by julianxhokaxhiu** [reply]
For AMD we suggest using DirectX 12 when using FFNx. Any other stack have shown problems including OpenGL. Unfortunately we can't do much as it is really dependand on the GPU drivers.

**Reply by malvarenga123** [reply]
@julianxhokaxhiu I've looked through BGFX's github issues on a few occasions and I couldn't find anything specific about AMD drivers that should render the game unplayable on anything but DX12. Where does this info come from and why is it only limited to FFNx? Have you guys tried to debug it with the BGFX team or even report it?
This really doesn't make a lot of sense to me as there are plenty of games that run perfectly on OpenGL, DX9, DX11 and Vulkan using these drivers. Games that are way more demanding than FF7.

**Reply by julianxhokaxhiu** [reply]
Yes I did try to approach the author and report the issues we face. Problem is that FF7 is an old game engine that uses the modern backend not really the way it is supposed to do. So I did try my best to make it work on top of an old approach. The result varies unfortunately based on GPUs and Drivers. The outcome is based on our own experience supporting every user out there, as you might have noticed the driver has been downloaded more than 200k times so far. But if you have experience in this area and you're willing to go down this path, I'll be more than happy to accept PRs :) Thanks and cheers!
